src/AutoRest.CSharp.V3/Output/Builders/ClientBuilder.cs

```diff
@@ -34,34 +34,72 @@ public static Client BuildClient(OperationGroup operationGroup)
             }
 
             string clientName = operationGroup.CSharpName();
-            List<Method> methods = new List<Method>();
-            List<Paging> pagingMethods = new List<Paging>();
+            Dictionary<string, OperationMethod> processedMethods = new Dictionary<string, OperationMethod>();
             foreach (Operation operation in operationGroup.Operations)
             {
                 Method? method = BuildMethod(operation, clientName, clientParameters);
                 if (method != null)
                 {
-                    methods.Add(method);
-                    var pageable = operation.Extensions.GetValue<IDictionary<object, object>>("x-ms-pageable");
-                    if (pageable != null)
+                    processedMethods.Add(operation.Language.Default.Name, new OperationMethod(operation, method));
+                }
+            }
+
+            List<Method> nextPageMethods = new List<Method>();
+            List<Paging> pagingMethods = new List<Paging>();
+            foreach (OperationMethod processed in processedMethods.Values)
+            {
+                IDictionary<object, object>? pageable = processed.Operation.Extensions.GetValue<IDictionary<object, object>>("x-ms-pageable");
+                if (pageable != null)
+                {
+                    //TODO: Assuming operationName is in this operation group: https://github.com/Azure/autorest.modelerfour/issues/85
+                    string? operationName = pageable.GetValue<string>("operationName")?.Split('_').Last();
+
+                    OperationMethod? next = null;
+                    if (operationName != null)
                     {
-                        Method nextPageMethod = BuildNextPageMethod(method);
-                        methods.Add(nextPageMethod);
-                        //TODO: This is a hack since we don't have the model information at this point
-                        var schemaForPaging = ((method.Response.ResponseBody as ObjectResponseBody)?.Type as SchemaTypeReference)?.Schema as ObjectSchema;
-                        Paging pagingMethod = GetClientMethodPaging(method, nextPageMethod, pageable, schemaForPaging);
-                        pagingMethods.Add(pagingMethod);
+                        if (!processedMethods.TryGetValue(operationName, out OperationMethod nextOperationMethod))
+                        {
+                            string processedOperationName = processed.Operation.Language.Default.Name;
+                            throw new Exception(
+                                $"The x-ms-pageable operationName \"{operationGroup.Key}_{operationName}\" for operation {operationGroup.Key}_{processedOperationName} was not found.");
```

Comment:
Use the original value of `operationName` extension here, no need to split and then re-join.

Reply:
Done.